File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/Mock.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.types.Product

/** Deterministic synthesized data for the hoist-args resolvers. */
internal object Mock {
    fun products(prefix: String, count: Int, offset: Int): List<Product> =
        (0 until count).map { i ->
            val n = offset + i
            Product(id = "$prefix-$n", name = "$prefix-$n")
        }
}

File: server_integration_tests/servers/typescript_server/src/impl/mock.ts
import { Product } from '../generated/types/product.js';

/** Deterministic synthesized data for the hoist-args resolvers. */
export const Mock = {
  products(prefix: string, count: number, offset: number): Product[] {
    return Array.from({ length: count }, (_, i) => {
      const n = offset + i;
      return { id: `${prefix}-${n}`, name: `${prefix}-${n}` };
    });
  },
};
